emcc.py

```diff
@@ -1811,104 +1811,97 @@ def include_and_export(name):
-            if shared.Settings.ASSERTIONS:
-              passes += ['--pass-arg=asyncify-asserts']
-            if shared.Settings.ASYNCIFY_IGNORE_INDIRECT:
-              passes += ['--pass-arg=asyncify-ignore-indirect']
-            else:
-              # if we are not ignoring indirect calls, then we must treat invoke_* as if
-              # they are indirect calls, since that is what they do - we can't see their
-              # targets statically.
-              shared.Settings.ASYNCIFY_IMPORTS += ['invoke_*']
-            # with pthreads we may call main through the __call_main mechanism, which can
-            # therefore reach anything in the program, so mark it as possibly causing a
-            # sleep (the asyncify analysis doesn't look through JS, just wasm, so it can't
-            # see what it itself calls)
-            if shared.Settings.USE_PTHREADS:
-              shared.Settings.ASYNCIFY_IMPORTS += ['__call_main']
-            # add the default imports
-            shared.Settings.ASYNCIFY_IMPORTS += DEFAULT_ASYNCIFY_IMPORTS
-
-            # return the full import name, including module. The name may
-            # already have a module prefix; if not, we assume it is "env".
-            def get_full_import_name(name):
-              if '.' in name:
-                return name
-              return 'env.' + name
-
-            shared.Settings.ASYNCIFY_IMPORTS = [get_full_import_name(i) for i in shared.Settings.ASYNCIFY_IMPORTS]
-
-            passes += ['--pass-arg=asyncify-imports@%s' % ','.join(shared.Settings.ASYNCIFY_IMPORTS)]
-
-            # shell escaping can be confusing; try to emit useful warnings
-            def check_human_readable_list(items):
-              for item in items:
-                if item.count('(') != item.count(')'):
-                  logger.warning('''emcc: ASYNCIFY list contains an item without balanced parentheses ("(", ")"):''')
-                  logger.warning('''   ''' + item)
-                  logger.warning('''This may indicate improper escaping that led to splitting inside your names.''')
-                  logger.warning('''Try to quote the entire argument, like this: -s 'ASYNCIFY_WHITELIST=["foo(int, char)", "bar"]' ''')
-                  break
-
-            if shared.Settings.ASYNCIFY_BLACKLIST:
-              check_human_readable_list(shared.Settings.ASYNCIFY_BLACKLIST)
-              passes += ['--pass-arg=asyncify-blacklist@%s' % ','.join(shared.Settings.ASYNCIFY_BLACKLIST)]
-            if shared.Settings.ASYNCIFY_WHITELIST:
-              check_human_readable_list(shared.Settings.ASYNCIFY_WHITELIST)
-              passes += ['--pass-arg=asyncify-whitelist@%s' % ','.join(shared.Settings.ASYNCIFY_WHITELIST)]
-          if shared.Settings.BINARYEN_IGNORE_IMPLICIT_TRAPS:
-            passes += ['--ignore-implicit-traps']
+        passes = ['--post-emscripten']
+        # safe heap must run before post-emscripten, so post-emscripten can apply the sbrk ptr
```

Comment:
this change broke the invariant mentioned in this comment (safe-heap must be first)